Metalama.Framework.Engine/Aspects/AspectDriver.cs

```diff
@@ -188,7 +189,10 @@ AspectInstanceResult CreateResultForError( Diagnostic diagnostic )
                 pipelineConfiguration.CodeFixFilter,
                 this._codeFixAvailability );
 
-            var userCodeInvoker = serviceProvider.GetRequiredService<UserCodeInvoker>();
+            // This is used for compilation aspects.
+            // Knowing that the aspect is applied to a compilation is not particularly useful when we want to understand dependencies between source files.
+            var allPredecessors = aspectInstance.Predecessors.Select( p => p.Instance ).SelectManyRecursive( x => x.Predecessors.Select( p => p.Instance ), includeRoot: true );
```

Comment:
This code is too poorly optimized; I would suggest to materialize as an `Immutable<SyntaxTree> PredecessorTreeClosure` probably on `IAspectPredecessorImpl`.